### ISSUE TITLE
Consider unifying the URL template syntax of callbacks and links

### ISSUE THREAD

**Issue by ePaul** [OAI/OpenAPI-Specification#998]
Both the [Link object](https://github.com/OAI/OpenAPI-Specification/blob/3.0.0-rc0/versions/3.0.md##linkObject) (in a response) and the [Callback object](https://github.com/OAI/OpenAPI-Specification/blob/3.0.0-rc0/versions/3.0.md#callbackObject) (in an operation) have a way of deriving a URL from some kind of pattern.

But those patterns are differently:

* For the `href` in the link object, we have a template string, which can contain `{$...}` expressions inside it. The full thing looks like `/users/{$request.body#/user/uuid}`.
* For the keys in the Callback object, we just have a single expression. So we have e.g. `$request.body#/failedUrl`.

Could we use the same syntax for both? This might also make the documentation less redundant.

**Comment by darrelmiller** [comment]
The `href` in the link object doesn't support variable expressions.  The `href` value is a constant value that is equal to some pathItem key.   However,  link parameter values do. And they are the same as those used in callback keys.

**Comment by darrelmiller** [comment]
I would argue they are the same syntax.  The only difference is in the `href` case the expression is embedded inside a parameter template and for callback it is not because the expression is supposed to evaluate to a complete URL and that URL is in the customer's domain so it isn't describable.

Having said this, I would argue that we should drop the first case completely.  So instead of having,

```YAML
components:
  links:
    UserRepositories:
      # returns array of '#/components/schemas/repository'
      href: '/2.0/repositories/{$response.body#/username}'
```
we do this.
```YAML
components:
  links:
    UserRepositories:
      # returns array of '#/components/schemas/repository'
      href: '/2.0/repositories/{name}'
      parameters:
        name: $response.body#/username
```

I don't like the idea of requiring tooling to match the href with expression to an existing operation.

/cc @OAI/tdc

**Comment by RobDolinMS** [comment]
TDC: @fehguy is good with what is done in the bottom example. 
TDC: @darrelmiller will make this change.

**Comment by webron** [comment]
Resolved by #1103!
